protocol/src/main/java/org/geysermc/mcprotocollib/protocol/data/game/entity/attribute/AttributeType.java

```diff
@@ -18,44 +19,44 @@ public interface AttributeType {
-        GENERIC_FLYING_SPEED("minecraft:generic.flying_speed", 0.4F, 0, 1024),
-        GENERIC_FOLLOW_RANGE("minecraft:generic.follow_range", 32, 0, 2048),
-        GENERIC_GRAVITY("minecraft:generic.gravity", 0.08, -1, 1),
-        GENERIC_JUMP_STRENGTH("minecraft:generic.jump_strength", 0.42, 0, 32),
-        GENERIC_KNOCKBACK_RESISTANCE("minecraft:generic.knockback_resistance", 0, 0, 1),
-        GENERIC_LUCK("minecraft:generic.luck", 0, -1024, 1024),
-        GENERIC_MAX_ABSORPTION("minecraft:generic.max_absorption", 0, 0, 2048),
-        GENERIC_MAX_HEALTH("minecraft:generic.max_health", 20, 1, 1024),
-        GENERIC_MOVEMENT_SPEED("minecraft:generic.movement_speed", 0.7F, 0, 1024),
-        GENERIC_SAFE_FALL_DISTANCE("minecraft:generic.safe_fall_distance", 3, -1024, 1024),
-        GENERIC_SCALE("minecraft:generic.scale", 1, 0.0625, 16),
-        ZOMBIE_SPAWN_REINFORCEMENTS("minecraft:zombie.spawn_reinforcements", 0, 0, 1),
-        GENERIC_STEP_HEIGHT("minecraft:generic.step_height", 0.6, 0, 10);
+        GENERIC_ARMOR(Key.key("minecraft:generic.armor"), 0, 0, 30),
+        GENERIC_ARMOR_TOUGHNESS(Key.key("minecraft:generic.armor_toughness"), 0, 0, 20),
+        GENERIC_ATTACK_DAMAGE(Key.key("minecraft:generic.attack_damage"), 2, 0, 2048),
+        GENERIC_ATTACK_KNOCKBACK(Key.key("minecraft:generic.attack_knockback"), 0, 0, 5),
+        GENERIC_ATTACK_SPEED(Key.key("minecraft:generic.attack_speed"), 4, 0, 1024),
+        PLAYER_BLOCK_BREAK_SPEED(Key.key("minecraft:player.block_break_speed"), 1, 0, 1024),
+        PLAYER_BLOCK_INTERACTION_RANGE(Key.key("minecraft:player.block_interaction_range"), 4.5, 0, 64),
+        PLAYER_ENTITY_INTERACTION_RANGE(Key.key("minecraft:player.entity_interaction_range"), 3, 0, 64),
+        GENERIC_FALL_DAMAGE_MULTIPLIER(Key.key("minecraft:generic.fall_damage_multiplier"), 1, 0, 100),
+        GENERIC_FLYING_SPEED(Key.key("minecraft:generic.flying_speed"), 0.4F, 0, 1024),
+        GENERIC_FOLLOW_RANGE(Key.key("minecraft:generic.follow_range"), 32, 0, 2048),
+        GENERIC_GRAVITY(Key.key("minecraft:generic.gravity"), 0.08, -1, 1),
+        GENERIC_JUMP_STRENGTH(Key.key("minecraft:generic.jump_strength"), 0.42, 0, 32),
+        GENERIC_KNOCKBACK_RESISTANCE(Key.key("minecraft:generic.knockback_resistance"), 0, 0, 1),
+        GENERIC_LUCK(Key.key("minecraft:generic.luck"), 0, -1024, 1024),
+        GENERIC_MAX_ABSORPTION(Key.key("minecraft:generic.max_absorption"), 0, 0, 2048),
+        GENERIC_MAX_HEALTH(Key.key("minecraft:generic.max_health"), 20, 1, 1024),
+        GENERIC_MOVEMENT_SPEED(Key.key("minecraft:generic.movement_speed"), 0.7F, 0, 1024),
+        GENERIC_SAFE_FALL_DISTANCE(Key.key("minecraft:generic.safe_fall_distance"), 3, -1024, 1024),
+        GENERIC_SCALE(Key.key("minecraft:generic.scale"), 1, 0.0625, 16),
+        ZOMBIE_SPAWN_REINFORCEMENTS(Key.key("minecraft:zombie.spawn_reinforcements"), 0, 0, 1),
+        GENERIC_STEP_HEIGHT(Key.key("minecraft:generic.step_height"), 0.6, 0, 10);
```

Comment:
Probably easier to add a constructor which handles the `Key.key()` rather than calling it so many times.

Reply:
Hm sure, but then I'd need to add @KeyPattern to the constructor arg to ensure they actually have the right name. Is that fine?

Reply:
Do you? It's not like it would ever be called outside of the class.

Reply:
Yeah or else I'd need to add a @SuppressWarnings("PatternValidation")

Reply:
In which case, yeah, fair enough.

Reply:
Alright